### ISSUE TITLE
Unstable dependencies

### ISSUE THREAD

**Issue by thiras** [ruqqus/ruqqus#496]
Issue tracker is **ONLY** used for reporting bugs.

Current requirements.txt is not reproducible for production deployment

## Expected Behavior
Get deterministic environment like Pipenv or Poetry provides

## Current Behavior
A broken pinned requirements.txt without deterministic versions

## Possible Solution
Implement poetry or pipenv

## Steps to Reproduce
Check the [issue](https://github.com/gevent/gevent/issues/1685) I've created at getenv.

**Comment by SuperLeaf1995** [comment]
use this for deploying

```sh
# server-specific keys
export HCAPTCHA_SECRET=1 CLOUDFLARE_KEY=1 AWS_ACCESS_KEY_ID=1 AWS_SECRET_ACCESS_KEY=1 \
PATREON_SECRET=1 PATREON_CLIENT_ID=1 PATREON_CLIENT_SECRET=1
# server stuff
export domain=localhost:8000
export REDIS_URL=redis://localhost:6379
export DATABASE_URL=postgres://localhost:5432/ruqqus
export MASTER_KEY=$(openssl rand -base64 32)
export PYTHONPATH=${PWD}
# load server
env gunicorn ruqqus.__main__:app -w 3 --preload --max-requests 5000 --max-requests-jitter 500
```

**Comment by thiras** [comment]
@SuperLeaf1995 it's not about environment variables but your requirements.txt. Please follow [this report](https://github.com/gevent/gevent/issues/1685) that CLEARLY STATES that greenlet and gunicorn version are incompatible;

```
You've installed greenlet 0.4.17, which changed sizes. If you keep using it, your application will likely crash. You should upgrade gevent to a version compatible with that greenlet release, 20.9. (Also, gevent 1.4 is no longer supported.)
```

As I reported in the link above here is the same output;

```
env gunicorn ruqqus.__main__:app -w 3 --preload --max-requests 5000 --max-requests-jitter 500
/usr/lib/python3.7/importlib/_bootstrap.py:219: RuntimeWarning: greenlet.greenlet size changed, may indicate binary incompatibility. Expected 144 from C header, got 152 from PyObject
  return f(*args, **kwds)
/usr/lib/python3.7/importlib/_bootstrap.py:219: RuntimeWarning: greenlet.greenlet size changed, may indicate binary incompatibility. Expected 144 from C header, got 152 from PyObject
  return f(*args, **kwds)
/usr/lib/python3.7/importlib/_bootstrap.py:219: RuntimeWarning: greenlet.greenlet size changed, may indicate binary incompatibility. Expected 144 from C header, got 152 from PyObject
  return f(*args, **kwds)
/usr/lib/python3.7/importlib/_bootstrap.py:219: RuntimeWarning: greenlet.greenlet size changed, may indicate binary incompatibility. Expected 144 from C header, got 152 from PyObject
  return f(*args, **kwds)
/usr/lib/python3.7/importlib/_bootstrap.py:219: RuntimeWarning: greenlet.greenlet size changed, may indicate binary incompatibility. Expected 144 from C header, got 152 from PyObject
  return f(*args, **kwds)
/usr/lib/python3.7/importlib/_bootstrap.py:219: RuntimeWarning: greenlet.greenlet size changed, may indicate binary incompatibility. Expected 144 from C header, got 152 from PyObject
  return f(*args, **kwds)
/usr/lib/python3.7/importlib/_bootstrap.py:219: RuntimeWarning: greenlet.greenlet size changed, may indicate binary incompatibility. Expected 144 from C header, got 152 from PyObject
  return f(*args, **kwds)
/usr/lib/python3.7/importlib/_bootstrap.py:219: RuntimeWarning: greenlet.greenlet size changed, may indicate binary incompatibility. Expected 144 from C header, got 152 from PyObject
  return f(*args, **kwds)
/usr/lib/python3.7/importlib/_bootstrap.py:219: RuntimeWarning: greenlet.greenlet size changed, may indicate binary incompatibility. Expected 144 from C header, got 152 from PyObject
  return f(*args, **kwds)
Segmentation fault
```

**Comment by captainmeta4** [comment]
Fix coming along with #577

**Comment by Butonix** [comment]
> use this for deploying
> 
> ```shell
> # server-specific keys
> export HCAPTCHA_SECRET=1 CLOUDFLARE_KEY=1 AWS_ACCESS_KEY_ID=1 AWS_SECRET_ACCESS_KEY=1 \
> PATREON_SECRET=1 PATREON_CLIENT_ID=1 PATREON_CLIENT_SECRET=1
> # server stuff
> export domain=localhost:8000
> export REDIS_URL=redis://localhost:6379
> export DATABASE_URL=postgres://localhost:5432/ruqqus
> export MASTER_KEY=$(openssl rand -base64 32)
> export PYTHONPATH=${PWD}
> # load server
> env gunicorn ruqqus.__main__:app -w 3 --preload --max-requests 5000 --max-requests-jitter 500
> ```

Tell me, what file should I create with this content?
